component/org.wso2.carbon.identity.dpop/src/main/java/org/wso2/carbon/identity/dpop/token/binder/DPoPBasedTokenBinder.java

```diff
@@ -100,6 +107,55 @@ public boolean isValidTokenBinding(Object request, String bindingReference) {
         return true;
     }
 
+    @Override
+    public boolean isValidTokenBinding(Object request, TokenBinding tokenBinding) {
+
+        try {
+            if (tokenBinding != null && DPoPConstants.OAUTH_DPOP_HEADER.equals(tokenBinding.getBindingType())) {
+
+                if (!((HttpServletRequest) request).getHeader(DPoPConstants.AUTHORIZATION_HEADER)
+                        .startsWith(DPoPConstants.OAUTH_DPOP_HEADER)) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("DPoP prefix is not defined correctly in the Authorization header.");
+                    }
+                    return false;
+                }
+
+                String dpopHeader = ((HttpServletRequest) request).getHeader(DPoPConstants.OAUTH_DPOP_HEADER);
+
+                if (StringUtils.isBlank(dpopHeader)) {
```

Comment:
have this validation logic in separate private methods

Reply:
Fixed